core/runtime/src/main/java/com/buschmais/jqassistant/core/runtime/api/configuration/JsonSchemaGenerator.java

```diff
@@ -0,0 +1,124 @@
+
+        SchemaGenerator generator = new SchemaGenerator(configBuilder.build());
+        ObjectNode schema = generator.generateSchema(clazz);
+        ObjectNode finalSchema = wrapJqassistant(schema);
+        saveSchemaToFile(finalSchema, path);
+        return finalSchema;
+    }
+
+    private static ObjectNode wrapJqassistant(ObjectNode schema) {
+        ObjectNode propertiesNode = JsonNodeFactory.instance.objectNode();
+        ObjectNode jqaWrapper = JsonNodeFactory.instance.objectNode();
+        ObjectNode definitionWrapper = JsonNodeFactory.instance.objectNode();
+        for (Map.Entry<String, JsonNode> property : schema.properties()) {
+            if (property.getKey().equals("properties")) {
+                propertiesNode.put("type", "object");
+                propertiesNode.set("properties", property.getValue());
+                propertiesNode.put("additionalProperties", false);
+            }
+            if (property.getKey().equals("$defs")) {
+                definitionWrapper.set("$defs", property.getValue());
+            }
+        }
+        jqaWrapper.set("jqassistant", propertiesNode);
+        definitionWrapper.put("type", "object");
+        definitionWrapper.set("properties", jqaWrapper);
+        return definitionWrapper;
+    }
+
+    private static void saveSchemaToFile(ObjectNode schema, String path) throws IOException {
+        File file = new File(path);
+        File parentDir = file.getParentFile();
+
+        if (parentDir != null && !parentDir.exists()) {
+            boolean directoriesCreated = parentDir.mkdirs();
+            if (directoriesCreated) {
+                System.out.println("Directory were created successfully");
+            } else {
+                System.out.println("Failed to create directory");
+            }
+        }
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.writerWithDefaultPrettyPrinter().writeValue(file, schema);
+        System.out.println("Schema saved: " + file.getAbsolutePath());
+    }
+
+    private static List<ResolvedType> getResolvedTypes(MethodScope target, ResolvedType resolvedType) {
+        if (resolvedType != null) {
+            if (resolvedType.getErasedType().equals(URI.class) || resolvedType.getErasedType().equals(File.class)) {
+                return List.of(target.getContext().resolve(String.class));
+            }
+            if (resolvedType.isInstanceOf(Map.class)) {
+                return List.of(target.getContext().resolve(Map.class));
+            }
+            if (resolvedType.getErasedType().equals(Optional.class)) {
+                return getResolvedTypes(target, resolvedType.getTypeParameters().get(0));
+            }
+        }
+        return List.of(target.getContext().resolve(resolvedType));
+    }
+
+    private static String mapToKebabCase(String name) {
+        if (name.matches("^[A-Z0-9_]+$")) {
```

Comment:
Can you check if Guava's CaseFormat.LOWER_CAMEL.to(CaseFormat.LOWER_HYPHEN, "myValue") can be used.

Reply:
Seems to work fine.